src/javascript/crypto/e2e/openpgp/pgpmime/utils.js

```diff
@@ -111,15 +120,204 @@ e2e.openpgp.pgpmime.Utils.parseHeaderValue = function(text) {
 
 
 /**
- * Splits a MIME message into lines.
- * @param {string} text The message to split
+ * Serialize a header into an array of strings.
+ * @param {e2e.openpgp.pgpmime.types.Header} header The header to serialize
  * @return {Array.<string>}
+ */
+e2e.openpgp.pgpmime.Utils.serializeHeader = function(header) {
+  var lines = [];
+  goog.object.forEach(header, function(headerValue, headerName) {
+    var line = [headerName + ': ' + headerValue.value];
+    if (goog.isDefAndNotNull(headerValue.params)) {
+      goog.object.forEach(headerValue.params, function(paramValue, paramName) {
+        line.push(paramName + '=' + goog.string.quote(paramValue));
+      });
+    }
+    // Content is wrapped at 64 chars, for compliance with the PEM protocol.
+    // Note that CRLF wrapped headers also require a trailing whitespace
+    lines.push(e2e.openpgp.pgpmime.Text.prettyTextWrap(
+        line.join('; '), constants.MimeNum.LINE_WRAP,
+        constants.Mime.CRLF + constants.Mime.WHITESPACE));
+  });
+  return lines;
+};
+
+
+/**
+ * Parses MIME headers into a dict, optionally extending existing headers.
+ * @param {string} text The MIME-formatted header.
+ * @param {?string=} opt_separator True if the node should be separated by
+ *     a different separator than crlf ('\r\n'). Some email systems only use
+ *     linefeeds or carriage returns.
+ * @return {e2e.openpgp.pgpmime.types.Header}
  * @private
  */
-e2e.openpgp.pgpmime.Utils.splitLines_ = function(text) {
-  return text.split(constants.Mime.CRLF);
+e2e.openpgp.pgpmime.Utils.parseHeader_ = function(text, opt_separator) {
+  var parsed = /** @type {e2e.openpgp.pgpmime.types.Header} */ ({});
+
+  // Implicit content-type is ASCII plaintext (RFC 2045).
+  goog.object.setIfUndefined(parsed, constants.Mime.CONTENT_TYPE, {
+    value: constants.Mime.PLAINTEXT,
+    params: {charset: constants.Mime.ASCII}
+  });
+  // Implicit content-transfer-encoding is 7bit (RFC 2045).
+  goog.object.setIfUndefined(parsed, constants.Mime.CONTENT_TRANSFER_ENCODING, {
+    value: constants.Mime.SEVEN_BIT
+  });
+
+  var headerLines = utils.splitHeaders_(text, opt_separator);
+  goog.array.forEach(headerLines, function(line) {
+    // Ex: 'Content-Type: multipart/encrypted'
+    var parts = line.split(':');
```

Comment:
Unless you can assure there will be no more colons, use goog.string.split with limit. That's also for all other uses of String.split below.